ch4/ucx: Suggest passing --without-go to ucx

When building the Arch Linux package for mpich-4.1.2 (which uses `--with-device=ch4:ucx`) on a system with go-1.21.4 installed, I get this build error.
```
make[4]: Entering directory '/home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx/bindings/go'
go env -w GO111MODULE=off ; \
cd /home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx/../../../modules/ucx/bindings/go/src/examples/perftest ;\
go build -o /home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx/bindings/go/.libs/tmp/goperftest
cd /home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx/../../../modules/ucx/bindings/go/src/ucx ;\
go build
# ucx
../../ucx/request.go:32:10: cannot define new methods on non-local type *C.ucp_request_param_t
# ucx
./request.go:32:10: cannot define new methods on non-local type *C.ucp_request_param_t
make[4]: *** [Makefile:645: goperftest] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: *** [Makefile:628: build] Error 1
make[4]: Leaving directory '/home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx/bindings/go'
make[3]: *** [Makefile:771: all-recursive] Error 1
make[3]: Leaving directory '/home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx'
make[2]: *** [Makefile:635: all] Error 2
make[2]: Leaving directory '/home/jed/src/aur/mpich/src/mpich-4.1.2/build/modules/ucx'
make[1]: *** [Makefile:31566: all-recursive] Error 1
make[1]: Leaving directory '/home/jed/src/aur/mpich/src/mpich-4.1.2/build'
make: *** [Makefile:7838: all] Error 2
```

I assume this is due to the vendored ucx being incompatible with latest go. (This error goes away if I uninstall go and rebuild.) But since MPICH doesn't need Go bindings, you might as well disable it.

--- a/src/mpid/ch4/netmod/ucx/subconfigure.m4
+++ b/src/mpid/ch4/netmod/ucx/subconfigure.m4
@@ -47,7 +47,7 @@ AM_COND_IF([BUILD_CH4_NETMOD_UCX],[
             else
                 ucx_opt_flags=""
             fi
-            PAC_CONFIG_SUBDIR_ARGS([modules/ucx],[--disable-static --enable-embedded --with-java=no $ucx_opt_flags],[],[AC_MSG_ERROR(ucx configure failed)])
+            PAC_CONFIG_SUBDIR_ARGS([modules/ucx],[--disable-static --enable-embedded --with-java=no --with-go=no $ucx_opt_flags],[],[AC_MSG_ERROR(ucx configure failed)])
             PAC_POP_ALL_FLAGS()
             ucxdir="modules/ucx"
         fi

